voc_builder/interactive.py

```diff
@@ -382,47 +382,49 @@ def handle_cmd_no() -> NoActionResult:
     if not choice:
         console.print('Skipped.', style='grey42')
         return NoActionResult(error='user_skip')
-
-    word_sample = WordSample(
-        word=choice.word,
-        word_normal=choice.word_normal,
-        word_meaning=choice.word_meaning,
-        pronunciation=choice.pronunciation,
+    
+    word_samplelist = []
+    for i in choice:
+        word_samplelist.append(WordSample(
+        word=i.word,
+        word_normal=i.word_normal,
+        word_meaning=i.word_meaning,
+        pronunciation=i.pronunciation,
         translated_text=ret.word_sample.translated_text,
         orig_text=ret.input_text,
-    )
-
-    try:
-        validate_result_word(word_sample, ret.input_text)
-    except WordInvalidForAdding as e:
-        console.print(f'Unable to add "{word_sample.word}", reason: {e}', style='grey42')
-        return NoActionResult(word=word_sample, stored_to_voc_book=False, error=str(e))
+    ))
+    
 
     word_store = get_word_store()
-    word_store.add(word_sample)
-    console.print(
-        (
-            f'[bold]"{word_sample.word}"[/bold] was added to your vocabulary book ([bold]{word_store.count()}[/bold] '
-            'in total), well done!\n'
-        ),
-        style='grey42',
-    )
+    for word_sample in word_samplelist:
+        try:
+            validate_result_word(word_sample, ret.input_text)
+        except WordInvalidForAdding as e:
+            console.print(f'Unable to add "{word_sample.word}", reason: {e}', style='grey42')
+            return NoActionResult(words=word_sample, stored_to_voc_book=False, error=str(e))
+
+        word_store.add(word_sample)
+        console.print(
+            (
+                f'[bold]"{word_sample.word}"[/bold] was added to your vocabulary book ([bold]{word_store.count()}[/bold] '
+                'in total), well done!\n'
+            ),
+            style='grey42',
+        )
 
     LastActionResult.trans_result = None
-    return NoActionResult(word=word_sample, stored_to_voc_book=True)
+    return NoActionResult(words=word_samplelist, stored_to_voc_book=True)
 
 
 class ManuallySelector:
     """A class dealing with manually word selection"""
 
     choice_skip = 'None of above, skip for now.'
 
-    def discard_word(self, word: WordSample):
-        """Discard a word added before"""
-        # Remove last word, mark as mastered
-        console.print(f'"{word.word}" was discarded, preparing other words...', style='grey42')
+    def init_selector(self, word: WordSample):
+        """Remove the last action word for prepare for the next action"""
+        # Remove last word
         get_word_store().remove(word.word)
-        get_mastered_word_store().add(word.word)
```

Comment:
Since the word.word will show up in prompt_select_word(). add to mastered word is kind of redundant now.